### PR TITLE
fix: fix get_all_groups

### PR DESCRIPTION
**Pull request by chasingrainbows**
## Description

This pull request updates the logic for fetching operator groups from the MetaRegistry contract to correctly handle group IDs as one-based (starting from 1) rather than zero-based. The associated tests have been updated to reflect this change and ensure proper coverage.

**Operator group indexing fix:**

* Updated `get_all_groups` in `meta_registry.py` to fetch groups using group IDs from 1 to `groupsCount` (inclusive), instead of starting from 0. This aligns with the contract's convention that group IDs are one-based and 0 is reserved.

**Test updates for new indexing:**

* Modified `test_contract_transformations.py` to remove assertions for group ID 0 and ensure that group fetching starts at 1, including checks for higher group IDs as appropriate. [[1]](diffhunk://#diff-88e2c22f238bf3b2c531bf575117faa7c85eca6743d6cd6b61e096e93493781fL866-R868) [[2]](diffhunk://#diff-88e2c22f238bf3b2c531bf575117faa7c85eca6743d6cd6b61e096e93493781fR879)
* Updated `test_meta_registry.py` integration test to expect the first group ID as 1 instead of 0.

## Related Issue/Task
- Related task: n/a
- Epic: srv3

## How Has This Been Tested?
Describe how you tested the changes:
- [x] Local tests (e.g., `pytest`)
- [ ] Manual testing (describe steps)
- [ ] Not tested (explain why)

## Checklist
- [ ] Documentation updated (if required)
- [ ] New tests added (if applicable)

### CODE REVIEW

File: tests/execution/contracts/test_contract_transformations.py [hunk 1]
```diff
@@ -863,9 +863,9 @@ def test_fetches_each_group_by_index(self):
         result = MetaRegistryContract.get_all_groups(contract, block_identifier="latest")
 
         assert result == groups
-        contract.get_operator_group.assert_any_call(0, "latest")
         contract.get_operator_group.assert_any_call(1, "latest")
         contract.get_operator_group.assert_any_call(2, "latest")
+        contract.get_operator_group.assert_any_call(3, "latest")
```
Comment (hweawer) [inline tests/execution/contracts/test_contract_transformations.py:868]: Consider adding:
```
 assert call(0, "latest") not in contract.get_operator_group.call_args_list
```

File: src/providers/execution/contracts/meta_registry.py [hunk 2]
```diff
@@ -92,5 +92,7 @@ def get_operator_group(self, group_id: int, block_identifier: BlockIdentifier) -
         return response
 
     def get_all_groups(self, block_identifier: BlockIdentifier) -> list[OperatorGroup]:
+        # Group IDs are one-based: the contract reserves NO_GROUP_ID = 0
+        # and creates real groups with IDs 1..groupsCount.
         all_groups = self.get_operator_groups_count(block_identifier)
-        return [self.get_operator_group(group_id, block_identifier) for group_id in range(all_groups)]
+        return [self.get_operator_group(group_id, block_identifier) for group_id in range(1, all_groups + 1)]
```
Comment (F4ever) [inline src/providers/execution/contracts/meta_registry.py:98]: Pls, test locally on Hoodi.
add test when groups is empty.

Comment (chasingrainbows) [reply src/providers/execution/contracts/meta_registry.py:98]: > add test when groups is empty
already exists
https://github.com/lidofinance/lido-oracle/blob/510e383ff701186fdf9e84422bcfa918703d09bb/tests/execution/contracts/test_contract_transformations.py#L848-L848

Comment (chasingrainbows) [reply src/providers/execution/contracts/meta_registry.py:98]: > Pls, test locally on Hoodi.

tested, 37 groups fetched

**Review by F4ever — APPROVED** [review]
